opentelemetry-otlp/src/exporter/http/trace.rs

```diff
@@ -46,7 +46,15 @@ impl SpanExporter for OtlpHttpClient {
         }
 
         Box::pin(async move {
-            client.send(request).await?;
+            let request_uri = request.uri().to_string();
+            let response = client.send(request).await?;
+
+            if !response.status().is_success() {
+                eprintln!(
+                    "OpenTelemetry export failed. Url: {}, Response: {:?}",
+                    request_uri, response
+                );
```

Comment:
Can we use [error_handler](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry/src/global/error_handler.rs#L51) here?

Reply:
+1. All errors, be routed via error_handler please!

Reply:
Thanks, I've pushed a fix and updated the description with an example output.

It has become a bit lengthy as 
* I didn't want to introduce allocation to store the `request_uri` in release mode, or
* Log raw response of arbitrary length from an external source in a production system.

So I've enabled this log only in debug mode.

If you'd prefer _some details_ without the request_uri like https://github.com/open-telemetry/opentelemetry-rust/pull/1498/commits/b6e30b9246ae841256951bcbd258d359f132a222 then let me know.